src/libstore/filetransfer.cc

```diff
@@ -824,6 +840,21 @@ void FileTransfer::download(FileTransferRequest && request, Sink & sink)
     }
 }
 
+template<typename... Args>
+FileTransferError::FileTransferError(FileTransfer::Error error, std::shared_ptr<string> response, const Args & ... args)
+    : Error(args...), error(error), response(response)
+{
+    const auto hf = hintfmt(args...);
+    // FIXME: Due to https://github.com/NixOS/nix/issues/3841 we don't know how
+    // to print different messages for different verbosity levels. For now
+    // we add some heuristics for detecting when we want to show the response.
```

Comment:
@edolstra We got stuck trying to make it respect `-vvv`, and opened up https://github.com/NixOS/nix/issues/3841 in response. Perhaps the heuristic with no way to override is good enough for now.